internal/handlers/common/compare.go

```diff
@@ -47,72 +48,92 @@ func compareScalars(a, b any) compareResult {
 	case float64:
 		switch b := b.(type) {
 		case float64:
+			if math.IsNaN(a) && math.IsNaN(b) {
+				return equal
```

Comment:
That's how MongoDB works? Interesting…

Reply:
Yes, if you are looking for NaN then you retrieve a document with a NaN value